test/integration/operator_policy_error_test.go

```diff
@@ -0,0 +1,126 @@
+	Ordered, Label("BVT"), func() {
+		const (
+			policyPath            = "../resources/operator_policy/test-op-err-initial.yaml"
+			correctPolicyPath     = "../resources/operator_policy/test-op-err-correct.yaml"
+			policyName            = "test-op-err-initial"
+			operatorPolicyName    = "test-op-err-initial"
+			subscriptionNamespace = "grcqeoptest-ns-43568"
+		)
+
+		BeforeAll(func() {
+			By("Create policy " + policyName + " which contains " + operatorPolicyName)
+			common.DoCreatePolicyTest(policyPath, common.GvrOperatorPolicy)
+		})
+
+		AfterAll(func() {
+			_, err := common.OcHub(
+				"delete", "-f", policyPath,
+				"-n", userNamespace, "--ignore-not-found=true",
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			_, err = common.OcManaged(
+				"delete", "ns", subscriptionNamespace, "--ignore-not-found=true",
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant due to invalid namespace", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(ContainSubstring("the operator namespace ('grcqeoptest-notcreated') does not exist"))
+		})
+
+		It("Namespace "+subscriptionNamespace+" should be created", func() {
+			_, err := common.OcManaged(
+				"create", "namespace", subscriptionNamespace,
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be patched with correct policy", func() {
+			_, err := common.OcHub(
+				"apply", "-f", correctPolicyPath, "-n", userNamespace,
+			)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be patched with incorrect source", func() {
+			_, err := common.OcHub(
+				"patch", "policies.policy.open-cluster-management.io", policyName,
+				"-n", userNamespace, "--type=json", "-p", `[{
+				"op":"replace", 
+				"path":"/spec/policy-templates/0/objectDefinition/spec/subscription/source",
+				"value":"redhat-operators-invalid"
+			}]`)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		It(operatorPolicyName+" should be NonCompliant due to incorrect source", func() {
+			By("Checking if the status of the operator policy is NonCompliant")
+			Eventually(
+				common.GetComplianceState(policyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(Equal(policiesv1.NonCompliant))
+			Eventually(
+				common.GetOpPolicyCompMsg(operatorPolicyName),
+				defaultTimeoutSeconds,
+				1,
+			).Should(ContainSubstring("CatalogSource 'redhat-operators-invalid' was not found"))
+		})
+
+		It(operatorPolicyName+" should be patched with correct policy", func() {
+			_, err := common.OcHub(
+				"apply", "-f", correctPolicyPath, "-n", userNamespace,
+			)
+			Expect(err).ToNot(HaveOccurred())
```

Comment:
We should add to check  it is compliant after the correct patch because the next test should be independent